src/engine/Bmc.cc

```diff
@@ -8,15 +8,20 @@
 #include "TermUtils.h"
 #include "TransformationUtils.h"
 
-VerificationResult BMC::solve(ChcDirectedGraph const & system) {
-    if (isTransitionSystem(system)) {
-        auto ts = toTransitionSystem(system, logic);
-        return solveTransitionSystem(*ts, system);
+VerificationResult BMC::solve(ChcDirectedGraph const & graph) {
+    if (isTransitionSystem(graph)) {
+        return solveTransitionSystem(graph);
     }
     return VerificationResult(VerificationAnswer::UNKNOWN);
 }
 
-VerificationResult BMC::solveTransitionSystem(TransitionSystem const & system, ChcDirectedGraph const & graph) {
+VerificationResult BMC::solveTransitionSystem(ChcDirectedGraph const & graph) {
```

Comment:
Why make 1 func into 2, can we just do line 19 and line 21 inside of the "solveTransitionSystemInternal", but it's Ok I think

It's just a function for 2 lines of code